### PR TITLE
Feat/stays components

### PR DESCRIPTION
**Pull request by bobetbat**
- https://github.com/windingtree/win/issues/13
- https://github.com/windingtree/win/issues/8

### CODE REVIEW

File: packages/stays/src/GlobalStyle.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { Grommet } from 'grommet';
+import { grommet } from 'grommet/themes';
+import { generate } from 'grommet/themes/base';
+import { deepMerge } from 'grommet/utils';
+import { ReactNode } from 'react';
+
+const baseTheme = deepMerge(grommet, {
+  ...generate(16),
```
Comment (kostysh) [inline packages/stays/src/GlobalStyle.tsx:8]: generation of the theme is useless for now. Anyway, we will create a custom theme later.
we have to keep the code as simple as possible

File: packages/stays/src/GlobalStyle.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  }
+});
+
+export const GlobalStyle: React.FC<{
+  children: ReactNode;
+}> = ({ children }) => {
+  return (
+    <Grommet
+      theme={baseTheme}
+      style={{
```
Comment (kostysh) [inline packages/stays/src/GlobalStyle.tsx:29]: any custom modifications added in context pf previous application must be removed

File: packages/stays/src/components/index.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { Button } from 'grommet';
+import styled from 'styled-components';
+
+const StyledButton = styled(Button)`
```
Comment (kostysh) [inline packages/stays/src/components/index.tsx:4]: usage of this ZOO of buttons must be revisited. In the new application, we have to remove ALL white, green, etc, buttons. We need to develop a standardized way of button creation based on the standard button (we will use themes).
Until this standard not developed please remove all custom stylings and just use grommets native buttons.

File: packages/stays/src/store/index.tsx [hunk 4]
```diff
@@ -44,7 +44,9 @@ const web3ModalConfig: Web3ModalConfig = {
     walletconnect: {
       package: WalletConnectProvider,
       options: {
-        [chainId]: rpc
+        rpc: {
```
Comment (kostysh) [inline packages/stays/src/store/index.tsx:47]: good catch!!! )))

**Review by kostysh — COMMENTED** [review]
Please implement fixes according to my recommendations in the comments.

File: packages/stays/src/GlobalStyle.tsx (hunk 1 above)
Comment (bobetbat) [reply packages/stays/src/GlobalStyle.tsx:8]: ✔️

File: packages/stays/src/components/Menu.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import { useMemo } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Menu as GMenu } from 'grommet';
+import { Menu as MenuIcon } from 'grommet-icons';
+import { useAppState } from '../store';
+import styled from 'styled-components';
+import { useWindowsDimension } from '../hooks/useWindowsDimension';
+import { pagesRoutesConfig, Routes } from '../Routes';
+
+const MenuWrap = styled(GMenu)`
```
[the remark this answers is not in the document]
Comment (bobetbat) [reply packages/stays/src/components/Menu.tsx:10]: ✔️

File: packages/stays/src/components/MessageBox.tsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
+import type { ReactNode } from 'react';
+import { useContext } from 'react';
+import { Box, Button, ResponsiveContext } from 'grommet';
+import { StatusInfo, Alert } from 'grommet-icons';
+
```
[the remark this answers is not in the document]
Comment (bobetbat) [reply packages/stays/src/components/MessageBox.tsx:5]: ✔️

File: packages/stays/src/GlobalStyle.tsx (hunk 2 above)
Comment (bobetbat) [reply packages/stays/src/GlobalStyle.tsx:29]: ✔️

File: packages/stays/src/components/index.tsx (hunk 3 above)
Comment (bobetbat) [reply packages/stays/src/components/index.tsx:4]: ✔️

File: packages/stays/src/pages/PageWrapper.tsx [hunk 7]
```diff
@@ -11,19 +13,21 @@ export const PageWrapper = ({ children }: PageWrapperProps) => {
   const { isConnecting, isRightNetwork } = useAppState();
 
   return (
-    <div>
+    <Box>
       <Header />
-      {isConnecting &&
-        <div>
+      <Box
+        margin={{ left: 'auto', right: 'auto', bottom: 'xlarge' }}
```
[the remark this answers is not in the document]
Comment (bobetbat) [reply packages/stays/src/pages/PageWrapper.tsx:19]: ✔️

File: packages/stays/src/store/index.tsx (hunk 4 above)
Comment (bobetbat) [reply packages/stays/src/store/index.tsx:47]: thanks 😃

File: packages/stays/src/pages/PageWrapper.tsx [hunk 8]
```diff
@@ -11,19 +13,21 @@ export const PageWrapper = ({ children }: PageWrapperProps) => {
   const { isConnecting, isRightNetwork } = useAppState();
 
   return (
-    <div>
+    <Box>
```
[the remark this answers is not in the document]
Comment (bobetbat) [reply packages/stays/src/pages/PageWrapper.tsx:16]: ✔️

File: packages/stays/src/components/Web3Modal.tsx [hunk 9]
```diff
@@ -1,35 +1,68 @@
...
+import { Login, Logout } from 'grommet-icons';
+import styled from 'styled-components';
+import { useWindowsDimension } from '../hooks/useWindowsDimension';
+
+const InnerSpinner = styled(Spinner)`
+  margin-left: 8px;
+`;
 
 export const SignInButton = () => {
+  const { winWidth } = useWindowsDimension();
```
[the remark this answers is not in the document]
Comment (bobetbat) [reply packages/stays/src/components/Web3Modal.tsx]: ✔️

**Review by kostysh — APPROVED** [review]
ready to merge

**Pull request merged**